src/Compilers/CSharp/Portable/Symbols/NativeIntegerTypeSymbol.cs

```diff
@@ -82,8 +94,92 @@ internal NativeIntegerTypeSymbol(NamedTypeSymbol underlying) : base(underlying,
 
         internal override NamedTypeSymbol AsNativeInt(bool asNativeInt) => asNativeInt ? this : _underlyingType;
 
-        internal override bool Equals(TypeSymbol t2, TypeCompareKind comparison, IReadOnlyDictionary<TypeParameterSymbol, bool> isValueTypeOverrideOpt = null) => _underlyingType.Equals(t2, comparison, isValueTypeOverrideOpt);
+        internal override bool Equals(TypeSymbol t2, TypeCompareKind comparison, IReadOnlyDictionary<TypeParameterSymbol, bool>? isValueTypeOverrideOpt = null) => _underlyingType.Equals(t2, comparison, isValueTypeOverrideOpt);
 
         public override int GetHashCode() => _underlyingType.GetHashCode();
+
+        private static ImmutableArray<Symbol> CalculateMembers(NativeIntegerTypeSymbol containingType, NamedTypeSymbol underlyingType)
+        {
+            var underlyingMembers = underlyingType.GetMembers();
+            var builder = ArrayBuilder<Symbol>.GetInstance();
+            addMethodIfAny(SpecialMember.System_Object__GetHashCode);
+            addMethodIfAny(SpecialMember.System_Object__Equals);
+            addMethodIfAny(SpecialMember.System_Object__ToString);
+            return builder.ToImmutableAndFree();
+
+            void addMethodIfAny(SpecialMember id)
+            {
+                var method = (MethodSymbol?)CSharpCompilation.GetRuntimeMember(
+                    underlyingMembers,
+                    SpecialMembers.GetDescriptor(id),
+                    CSharpCompilation.SpecialMembersSignatureComparer.Instance,
+                    accessWithinOpt: null);
+                if (method is object)
+                {
+                    builder.Add(new NativeIntegerMethodSymbol(containingType, method));
```

Comment:
>new NativeIntegerMethodSymbol(containingType, method) [](start = 32, length = 53)

Where do we make sure the methods are unified with underlying methods for the purpose of IL generation?

Reply:
Same for types, for example ```typeof(nint)```. 

---
In reply to: [382302745](https://github.com/dotnet/roslyn/pull/41728#discussion_r382302745) [](ancestors = 382302745)

Reply:
I think it would be good to find a good place to assert that Cci layer never sees instances of these synthetic symbols, both types and members.

---
In reply to: [382303254](https://github.com/dotnet/roslyn/pull/41728#discussion_r382303254) [](ancestors = 382303254,382302745)